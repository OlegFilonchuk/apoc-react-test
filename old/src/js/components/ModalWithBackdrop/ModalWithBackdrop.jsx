import React from 'react';
import PropTypes from 'prop-types';

import withBackdrop from '../withBackdrop/withBackdrop';
import Modal from '../Modal/Modal';

const ExtendedModal = withBackdrop(Modal, 'isOpen');

const ModalWithBackdrop = props => <ExtendedModal {...props} backdrop={{ onClick: props.onOutsideModalClick }} />;

ModalWithBackdrop.propTypes = {
  onOutsideModalClick: PropTypes.func
};

ModalWithBackdrop.defaultProps = {
  onOutsideModalClick: () => true
};

export default ModalWithBackdrop;
