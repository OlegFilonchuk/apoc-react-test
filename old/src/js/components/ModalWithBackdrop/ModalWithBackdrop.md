This component has same properties set as [`<Modal/>`](http://localhost:1234/#Modal).
The difference is that it has a [`<Backdrop/>`](http://localhost:1234/#Backdrop) attached

    function openModal() {
      setState({isOpen: true})
    }

    function closeModal() {
      setState({isOpen: false})
    }

    (
    <div>
      <Button className="sc-btn-default" onClick={openModal}>Open Modal Dialog</Button>
      <ModalWithBackdrop onOutsideModalClick={closeModal} isOpen={state.isOpen} header="Modal Dialog" footer={<Button className="sc-btn-default" onClick={closeModal}>Close</Button>}>
        Hello World!
      </ModalWithBackdrop>
    </div>
    )