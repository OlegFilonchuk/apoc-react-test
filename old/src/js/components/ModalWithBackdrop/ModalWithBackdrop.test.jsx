import React from 'react';
import { shallow, mount } from 'enzyme';

import ModalWithBackdrop from './ModalWithBackdrop';

describe('<ModalWithBackdrop />', () => {
  describe('ModalWithBackdrop render', () => {
    it('should render ModalWithBackdrop', () => {
      const modal = (
        <ModalWithBackdrop header="" footer={<span>test</span>}>
          Hello World from fist modal!
        </ModalWithBackdrop>
      );

      const modalDialog = shallow(modal);

      expect(modalDialog).toMatchSnapshot();
    });

    it('should call onOutsideModalClick() callback when user clicked outside modal dialog', () => {
      // arrange
      const onOutsideModalClick = jest.fn();

      // act
      const component = mount(<ModalWithBackdrop isOpen onOutsideModalClick={onOutsideModalClick} />);

      component.find('.backdrop').simulate('click');
      expect(onOutsideModalClick).toBeCalled();
    });
  });
});
