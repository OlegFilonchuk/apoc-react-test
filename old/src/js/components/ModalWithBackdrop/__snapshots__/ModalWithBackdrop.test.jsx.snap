// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ModalWithBackdrop /> ModalWithBackdrop render should render ModalWithBackdrop 1`] = `
<ModalWithBackdrop
  backdrop={
    Object {
      "onClick": [Function],
    }
  }
  footer={
    <span>
      test
    </span>
  }
  header=""
  onOutsideModalClick={[Function]}
>
  Hello World from fist modal!
</ModalWithBackdrop>
`;
